### PR TITLE
Added CLI to quickly sort or verify version numbers.

### PR DESCRIPTION
**Pull request by venthur**
Closes: #111

```bash
usage: version.py [-h] [--sort version [version ...] | --verify version
                  [version ...]]

optional arguments:
  -h, --help            show this help message and exit
  --sort version [version ...]
                        Return sorted list of versions.
  --verify version [version ...]
                        Exit with 0 if versions are sorted, or 1 otherwise.
```

Example output:

```bash
$ python -m packaging.version --sort 1 2 324 234.2 0.beta 0.alpha
0a0 0b0 1 2 234.2 324
```

### CODE REVIEW

**Comment by venthur** [comment]
Enforcing 100% code coverage might be a bit too strict here.

This patch [introduces](https://github.com/pypa/packaging/pull/183/files#diff-832b3493370c7de79eff39a50afebf2aR464) a code block that cannot (easily) be tested with pytest:

```python
if __name__ == "__main__":
    sys.exit(main())
```

As you'll see, I already put all the code into the `main` function so i can test them properly, but this little block above remains uncovered. If you insist on 100% coverage, please tell me how to execute this using pytest+coverage across all supported Python versions.

Cheers!

**Comment by di** [comment]
You can test this code block by moving the `__name__` check into the `main` function instead, and monkeypatching `__name__`:

```python
def main(args=None):
    if __name__ == "__main__":
        parser = argparse.ArgumentParser()
        ...
        # return with exitcode zero if list was sorted or 1 otherwise
        sys.exit(versions != sorted_versions)

main()
```

(Also note the improvement on the last line here)

**Comment by venthur** [comment]
@di The moving `__name__` into `main` would work, but is not very pythonic. You sure you want that change?

**Comment by pradyunsg** [comment]
FWIW, if that's how we do it, I'd rather that you return early instead of an additional indentation in the entire body.

As for the style, yea, I don't like it either. Just add a #pragma no cover there. That is okay.

File: packaging/version.py [hunk 1]
```diff
@@ -418,3 +420,46 @@ def _cmpkey(epoch, release, pre, post, dev, local):
...
+
+        # return with exitcode zero if list was sorted or 1 otherwise
+        if versions == sorted_versions:
+            return 0
+        else:
+            return 1
+
+
+if __name__ == "__main__":  # pragma: no cover
+    sys.exit(main())
```
Comment (pradyunsg) [inline packaging/version.py]: You'd probably want the pragma on this line, and not the previous one.

**Comment by venthur** [comment]
Any updates on this PR?

**Comment by deveshks** [comment]
Hi @venthur 

Is this something you are currently working on? If not, do you mind If I pick it up?

**Comment by brettcannon** [comment]
This PR seems to have stalled, so I'm going to close it (plus I'm not sure if this is something we specifically want; will comment on the issue).

**Pull request closed without merging**